app/scripts/controllers/transactions.js

```diff
@@ -161,9 +161,11 @@ module.exports = class TransactionController extends EventEmitter {
     this.emit(`${txMeta.id}:unapproved`, txMeta)
   }
 
-  async newUnapprovedTransaction (txParams) {
+  async newUnapprovedTransaction (txParams, opts = {origin: 'metamask'}) {
```

Comment:
seems like a dangerous default - also should not be necesary

we set it here
https://github.com/MetaMask/metamask-extension/blob/0582fdb617a23cc56e9e08460f09d86769599fc1/app/scripts/background.js#L157